### ISSUE TITLE
data.table setup error at CI macOS job in master

### ISSUE THREAD

**Issue by StrikerRUS** [microsoft/LightGBM#4008]
```
  There is a binary version available but the source version is later:
           binary source needs_compilation
data.table 1.13.6 1.14.0              TRUE


...


* installing *source* package ‘data.table’ ...
** package ‘data.table’ successfully unpacked and MD5 sums checked
** using staged installation
zlib 1.2.11 is available ok
R CMD SHLIB supports OpenMP without any extra hint
** libs
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG   -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -fopenmp  -fPIC  -Wall -g -O2  -c assign.c -o assign.o
clang: error: unsupported option '-fopenmp'
make: *** [assign.o] Error 1
ERROR: compilation failed for package ‘data.table’
* removing ‘/Users/runner/Rlib/data.table’

The downloaded source packages are in
	‘/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/RtmpVQXmzz/downloaded_packages’
Warning message:
In install.packages(c("data.table", "jsonlite", "Matrix", "R6",  :
  installation of package ‘data.table’ had non-zero exit status
```

Full logs: https://github.com/microsoft/LightGBM/runs/1946047294?check_suite_focus=true.

**Comment by StrikerRUS** [comment]
cc @jameslamb

**Comment by StrikerRUS** [comment]
I guess we just should wait for compiled binary version from CRAN.

**Comment by jameslamb** [comment]
it is pretty strange, since only the R 3.6 jobs are failing, even though Mac binaries for the newest `{data.table}` version are not available for R 4.0.

![image](https://user-images.githubusercontent.com/7608904/108636500-3675ef80-744b-11eb-9bfa-8cd176519195.png)

https://cran.r-project.org/web/packages/data.table/index.html

**Comment by jameslamb** [comment]
I think this issue will resolve itself when the `{data.table}` binaries are up, but I've also opened #4010 with a proposal to prevent this in the future. I was able to reproduce the issue on my Mac and fix it with the proposal in that PR.
